opal/class/opal_lifo.h

```diff
@@ -104,8 +104,47 @@ static inline bool opal_lifo_is_empty( opal_lifo_t* lifo )
     return (opal_list_item_t *) lifo->opal_lifo_head.data.item == &lifo->opal_lifo_ghost;
 }
 
+#if OPAL_HAVE_ATOMIC_LLSC_PTR
 
-#if OPAL_HAVE_ATOMIC_CMPSET_128
+static inline opal_list_item_t *opal_lifo_push_atomic (opal_lifo_t *lifo,
+                                                       opal_list_item_t *item)
+{
+    opal_list_item_t *next;
+
+    do {
+        item->opal_list_next = next = (opal_list_item_t *) opal_atomic_ll_ptr (&lifo->opal_lifo_head.data.item);
+        opal_atomic_wmb ();
```

Comment:
I think it's pretty unconventional to have a barrier in the middle of a `lwarx`/`stwcx.` pair.  The "third party" memory store to `item->opal_list_next` (which might clear the "reservation") and the wmb (which could also potentially mess with the reservation) are disconcerting.

Do you have a trustworthy source example of this style of unconventional `lwarx`/`stwcx.` usage?